### PR TITLE
Add v2.4.0 breaking change documentation

### PR DESCRIPTION
**Pull request by matthchr**
* Describes how to use asoctl clean crds to remove beta resources.

**If applicable**:
- [x] this PR contains documentation
- [ ] this PR contains tests

### CODE REVIEW

File: docs/hugo/content/guide/breaking-changes/breaking-changes-v2.4.0.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+1. If you have never installed a `beta` version of ASOv2, or used an `beta` version of a CRD, **no action is required**.
+   If you're not sure, run the below steps anyway. They will not do anything if beta was never used.
+2. Ensure the cluster is running the `v2.3.0` version of ASO.
+3. Install the `v2.4.0` version of the `asoctl` tool by following the [installation instructions]( {{< relref "asoctl#installation" >}})
+4. Run `asoctl clean crds`. This will migrate your resources to a newer version if needed and remove the `beta` CRD
+   versions from the CRD `Status.StoredVersions` collection. For more details on this command see the
+   [asoctl clean crds documentation]( {{< relref "asoctl#clean-crds" >}} )
+
+Once you have successfully executed the above steps, you can upgrade to `v2.4.0`. When upgrading to `v2.4.0` the following
+error means that you need to run the above steps:
```
Comment (theunrepentantgeek) [inline docs/hugo/content/guide/breaking-changes/breaking-changes-v2.4.0.md:20]: These two sentences belong in different paragraphs.

```suggestion
Once you have successfully executed the above steps, you can upgrade to `v2.4.0`. 

When upgrading to `v2.4.0` the following error means that you need to run the above steps:
```

File: docs/hugo/content/guide/breaking-changes/breaking-changes-v2.4.0.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+```
+"msg"="failed to apply CRDs" "error"="failed to apply CRD storageaccountsqueueservicesqueues.storage.azure.com: CustomResourceDefinition.apiextensions.k8s.io \"storageaccountsqueueservicesqueues.storage.azure.com\" is invalid: status.storedVersions[0]: Invalid value: \"v1beta20210401storage\": must appear in spec.versions" 
+```
```
Comment (theunrepentantgeek) [inline docs/hugo/content/guide/breaking-changes/breaking-changes-v2.4.0.md:24]: Should you mention where the error may be observed?

**Review by theunrepentantgeek — APPROVED** [review]

**Review by super-harsh — APPROVED** [review]

File: docs/hugo/content/guide/breaking-changes/breaking-changes-v2.4.0.md (hunk 1 above)
Comment (matthchr) [reply docs/hugo/content/guide/breaking-changes/breaking-changes-v2.4.0.md:20]: Good catch

File: docs/hugo/content/guide/breaking-changes/breaking-changes-v2.4.0.md (hunk 2 above)
Comment (matthchr) [reply docs/hugo/content/guide/breaking-changes/breaking-changes-v2.4.0.md:24]: Done

**Pull request merged**